### DISCUSSION TITLE
Is it possible for someone to have permission to invoke the execute function as an attack

### DISCUSSION THREAD

**Discussion by BradMoonUESTC** [PaulRBerg/prb-proxy#48] [Q&A] [answered]
When PRBProxy contract owner appoint you can execute some function, I think it is possible that you can use an malicious target in `function execute(address target, bytes calldata data)` which can change the `mapping(address => mapping(address => mapping(bytes4 => bool))) internal permissions;`  through delegatecall
Is such behavior allowed？

**Comment by PaulRBerg** [comment, marked as answer]
Yep, this is possible, but it's just a natural consequence of the power of `DELEGATECALL`. You should note, however, that the `execute` does not allow any target contract to change the `owner`:

https://github.com/paulrberg/prb-proxy/blob/0cab8248a4c513fa86e4064c352cff054d54ff90/contracts/PRBProxy.sol#L91-L94

But besides that, the target can do whatever it wants to the proxy. That's by design.

Also - if indeed the target was malicious, why would it take a detour and approve another malicious target? Why would it not peform the attack itself?

Bottom line is - users just should be very careful what target contracts they interact with via PRBProxy. Just like they should be very careful what contracts they approve to spend their ERC-20 token allowances.
